components/builder-depot/src/server.rs

```diff
@@ -461,32 +473,34 @@ fn download_latest_origin_encryption_key(req: &mut Request) -> IronResult<Respon
         OriginPublicEncryptionKey,
     >(req, &request) {
         Ok(key) => key,
-        Err(err) => return Ok(render_net_error(&err)),
+        Err(err) => {
+            if err.get_code() == ErrCode::ENTITY_NOT_FOUND {
+                generate_origin_encryption_keys(&origin, req)?;
+                return download_latest_origin_encryption_key(req);
```

Comment:
Not a fan of a recursive call here - would be better to make this linear - generate_origin_encryption_keys could return the keys, not an IronResponse.